File: cs/lc6.cs
public class Solution {
    public string Convert(string s, int numRows) {
        List<List<char>> ans = new List<List<char>>();
        int idx = 0;
        for(int i = 0;i< Math.Min(numRows, s.Length);i++)
        {
            ans.Add(new List<char>());
        }
        
        while(idx < s.Length)
        {
            for(int i = 0;i< numRows;i++)
            {
                if(idx < s.Length)
                {
                    ans[i].Add(s[idx]);
                    idx++;
                }
                else
                    break;
            }
            for(int i = numRows-2; i>0; i--)
            {
                if(idx < s.Length)
                {
                    ans[i].Add(s[idx]);
                    idx++;
                }
                else
                    break;
            }
        }
        
        StringBuilder sb = new StringBuilder();
        foreach(var l in ans)
        {
            sb.Append(String.Join("",l));
        }
        return sb.ToString();
    }
}

File: python/lc6.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        ans = []
        idx = 0
        for i in range(numRows):
            if idx < len(s):
                ans.append([s[idx]])
                idx += 1
            else:
                break
        while(idx < len(s)):
            for i in range(numRows-2, 0,-1):
                if idx < len(s):
                    ans[i].append(s[idx])
                    idx += 1
                else:
                    break
            for i in range(numRows):
                if idx < len(s):
                    ans[i].append(s[idx])
                    idx += 1
                else:
                    break
        # print(ans)
        return ''.join(''.join(a) for a in ans)
